vignettes/archetypes.Rmd

```diff
@@ -0,0 +1,258 @@
+#>  1                1                0                0                0
+#>  2                1                1                0                0
+#>  3                1                1                1                0
+#>  4                1                0                0                0
+#>  5                1                1                0                0
+#>  6                1                1                1                0
+#>  7                1                0                0                0
+#>  8                1                1                0                0
+#>  9                1                1                1                0
+#> 10                1                0                0                0
+#> # ℹ 590 more rows
+#> # ℹ 19 more variables: x_group_2_time_3 <dbl>, x_group_2_time_4 <dbl>,
+#> #   nuisance_biomarker1 <dbl>, nuisance_biomarker2 <dbl>,
+#> #   nuisance_status1_absent <dbl>, nuisance_status2_present <dbl>,
+#> #   nuisance_baseline <dbl>, nuisance_baseline.timetime_2 <dbl>,
+#> #   nuisance_baseline.timetime_3 <dbl>, patient <chr>, time <chr>,
+#> #   change <dbl>, missing <lgl>, baseline <dbl>, group <chr>, …
+```
+
+Those new columns constitute a custom model matrix to describe the desired parameterization. We have effects of interest to express successive differences,
+
+
+```r
+attr(archetype, "brm_archetype_interest")
+#> [1] "x_group_1_time_2" "x_group_1_time_3" "x_group_1_time_4"
+#> [4] "x_group_2_time_2" "x_group_2_time_3" "x_group_2_time_4"
+```
+
+and we have nuisance variables. Some nuisance variables are continuous covariates, while others are levels of one-hot-encoded concomitant factors or interactions of those concomitant factors with baseline and/or subgroup. All nuisance variables are centered at their means so the reference level of the model is at the "center" of the data and not implicitly conditional on a subset of the data. In addition, some nuisance variables are automatically be dropped in order to ensure the model matrix is full-rank. This is critically important to preserve the interpretation of the columns of interest and make sure the informative priors behave as expected.
+
+
+```r
+attr(archetype, "brm_archetype_nuisance")
+#> [1] "nuisance_biomarker1"          "nuisance_biomarker2"         
+#> [3] "nuisance_status1_absent"      "nuisance_status2_present"    
+#> [5] "nuisance_baseline"            "nuisance_baseline.timetime_2"
+#> [7] "nuisance_baseline.timetime_3"
+```
+
+The factors of interest linearly map to marginal means. To see the mapping, call `summary()` on the archetype. The printed output helps build intuition on how the archetype is parameterized and what those parameters are doing.
+
+
+```r
+summary(archetype)
+#> # This tibble is an informative prior archetype in brms.mmrm.
+#> # The fixed effect parameters of interest express the
+#> # marginal means as follows (on the link scale):
```

Comment:
Yes, I was thinking ahead to #87.

I used the term "link scale" because it seems to refer to the scale of the parameters: https://stats.stackexchange.com/questions/430855/should-modeled-means-in-glmm-be-averaged-on-link-or-response-scale. If we have a model with `g(E(y)) = X * beta` for link function `g()`, then the "link scale" is the scale of `X * beta`, and the "response scale" is the scale of `g^{-1}(X * beta)`. Please let me know if this phrasing is not clear.

Reply:
Actually, non-Gaussian families would be extremely difficult to add to the package (c.f. https://github.com/openpharma/brms.mmrm/discussions/106#discussioncomment-9499513). So I don't think it makes sense to add documentation for something that is a long ways off, if we even consider it in scope. Removed "(on the link scale)" in eb398805e35327ee287bcf5aafcbb1add8ec1b4b.

Reply:
Thanks. Makes sense.